web/src/components/dashboard/SidebarCardFundingNudge.test.tsx

```diff
@@ -46,8 +82,68 @@ describe("<SidebarCardFundingNudge />", () => {
     card = generateSidebarCardContent({ id: fundingNudge });
   });
 
+  const generateTaxFilingUserData = (params: {
+    publicFiling: boolean;
+    formedInNavigator?: boolean;
+    businessName?: string;
+    responsibleOwnerName?: string;
+    taxId?: string;
+  }): UserData => {
+    let formationData: FormationData = generateEmptyFormationData();
+
+    const legalStructureId = params.publicFiling
+      ? randomPublicFilingLegalType()
+      : randomLegalStructure({ requiresPublicFiling: false }).id;
+    if (params.publicFiling) {
+      formationData = generateFormationData(
+        {
+          completedFilingPayment: !!params.formedInNavigator,
+          getFilingResponse: generateGetFilingResponse({ success: params.formedInNavigator }),
+        },
+        legalStructureId as FormationLegalType
+      );
+    }
+
+    return generateUserDataForBusiness(
+      generateBusiness({
+        profileData: generateProfileData({
+          legalStructureId: legalStructureId,
+          operatingPhase: randomElementFromArray(
+            OperatingPhases.filter((obj) => {
+              return obj.displayTaxAccessButton;
+            })
+          ).id,
+          taxId: params.taxId ? `*${params.taxId.slice(1)}` : "",
+          encryptedTaxId: params.taxId ? `encrypted-${params.taxId}` : "",
+          businessName: params.businessName || "",
+          responsibleOwnerName: params.responsibleOwnerName || "",
+        }),
+        formationData: formationData,
+        taxFilingData: generateTaxFilingData({ state: undefined }),
+      })
+    );
+  };
```

Comment:
[sand] If we're using this exact same function both here and in `TaxAccessStepTwo.test.tsx`, this feels like a good candidate to move to a centralized location so that it can be imported and used in both places.

Reply:
It's actually not necessary in this particular test, so removed.